### PR TITLE
Add Japanese translation

### PR DESCRIPTION
**Pull request by kozo2**
Translate 2 terms (Japanese)

Fill in each of the sections (using NA for those that are not applicable).

## Author: 

- @kozo2

## Language: 

- Japanese

## Terms translated:

- operating_system
- filesystem

### CODE REVIEW

File: glossary.yml [hunk 1]
```diff
@@ -4593,6 +4599,12 @@
       A program that provides a standard interface to whatever hardware it is running
       on. Theoretically, any program that only interacts with the operating system
       should run on any computer that operating system runs on.
+  ja:
+    term: "オペレーティングシステム"
+    def: >
+      動作させるハードウェアがどんなものでも標準的なインターフェイスを提供するプログラム。
```
Comment (masamiy) [inline glossary.yml:4605]:
```suggestion
      動作させるハードウェアがどのようなものであっても標準的なインターフェイスを提供するプログラム。
```

File: glossary.yml [hunk 2]
```diff
@@ -4593,6 +4599,12 @@
...
+      理論的には、オペレーティングシステムとしかやりとりを行わないプログラムはどれでも、
+      オペレーティングシステムが動いているどんなコンピューターでも動かなければならない。
```
Comment (masamiy) [inline glossary.yml:4607]:
```suggestion
      そのオペレーティングシステムが搭載されているいかなるコンピューター上でも動作するはずである。
```

File: glossary.yml [hunk 3]
```diff
@@ -2680,6 +2680,12 @@
       La parte del sistema operativo que administra cómo se almacenan y recuperan
       los archivos. También se usa para referirse a todos esos archivos y directorios
       o a la forma específica en que se almacenan (como en "el sistema de archivos Unix").
+  ja:
+    term: "ファイルシステム"
+    def: >
+      ファイルの保存方法と取得方法を管理する[オペレーティングシステム](#operating_system)の部分。
```
Comment (masamiy) [inline glossary.yml:2686]:
```suggestion
      [オペレーティングシステム](#operating_system)におけるファイルの保存方法と取得方法を管理する部分。
```

**Review by masamiy — CHANGES_REQUESTED** [review]
Thanks @kozo2 for your translation work. I made some comments as some expressions were a little colloquial (to me).

File: glossary.yml [hunk 4]
```diff
@@ -2680,6 +2680,12 @@
...
+      また、それらすべてのファイルとディレクトリを参照するのに用いられるもの、
```
[the remark this answers is not in the document]
Comment (kozo2) [reply glossary.yml:2687]: I will ask Carpentries Slack if there is a rule for it.

**Review by naoe-tatara — APPROVED** [review]
Thanks @kozo2 , I think the translation looks good :)

**Comment by fmichonneau** [comment]
@masamiy could you please re-review and merge if you are happy with the changes? Thanks!

**Comment by elletjies** [comment]
@all-contributors please add @kozo2  for translation

**Pull request merged**